traitsui/extras/demo.py

```diff
@@ -586,6 +615,13 @@ def _get_path(self):
 
         return path
 
+    def _get_base_url(self):
+        if isdir(self.path):
+            base_dir = self.path
+        else:
+            base_dir = dirname(self.path)
+        return base_dir
```

Comment:
I had punted on working out the behaviour of absolute vs. relative paths.  Personally I think the correct behaviour is that the main script should be passing an absolute path to its children and should not be accepting any command-line paths.

Reply:
OK. That works too. We should fix `tutor.py` then.
(Wondering why we have two entry points to this app)

Reply:
> We should fix tutor.py then.

Or maybe the `demo` here to always convert the directory path to absolute path early.
This is getting more orthogonal and could be fixed in a separate PR.

Reply:
There are two entry points because before this year there were two completely different sources: this `demo.py` and a separate `tutor.py` that included a completely parallel and differently architected demo application.

This is now a merging of the two codebases so that it can handle both sets of examples.  At some point the traits demo and tutor examples should be merged (with likely a lot of clean-up!)

And yes, there is a lot of work needed on this codebase to get it where we want it.  Some of it should be in different PRs.

Reply:
Thank you!  Now I understand why there are two and that they should be merged. I will let you decide if you'd like to handle the absolute path issue here or separately.
Resolving this thread.